app/src/main/java/com/example/myfitnessbuddy/activities/RegistrationActivity.java

```diff
@@ -1,253 +1,109 @@
-        viewPager.setAdapter(pagerAdapter);
-        viewPager.addOnPageChangeListener(new TabLayout.TabLayoutOnPageChangeListener(tabLayout));
-        viewPager.setAllowedSwipeDirection(none);
-
-        tabLayout.addOnTabSelectedListener(
-                new TabLayout.ViewPagerOnTabSelectedListener(viewPager) {
-                    @Override
-                    public void onTabSelected(TabLayout.Tab tab) {
-                        super.onTabSelected(tab);
-                        int numTab = tab.getPosition();
-                        viewPager.setCurrentItem(numTab);
-                    }
-                });
-
+    protected int getActivityLayout() {
+        return R.layout.activity_registration;
     }
 
-    private void sendVerificationCode() {
-        Log.i("Noemi","Phone number verification started");
-        //Phone number validation needed here....
-        String phoneNumber = user.getPhoneNumber();
-
-        PhoneAuthProvider.getInstance().verifyPhoneNumber(
-                phoneNumber,        // Phone number to verify
-                60,                 // Timeout duration
-                TimeUnit.SECONDS,   // Unit of timeout
-                this,               // Activity (for callback binding)
-                mCallbacks);        // OnVerificationStateChangedCallbacks
-    }
-
-    PhoneAuthProvider.OnVerificationStateChangedCallbacks mCallbacks = new PhoneAuthProvider.OnVerificationStateChangedCallbacks() {
-        @Override
-        public void onVerificationCompleted(@NonNull PhoneAuthCredential phoneAuthCredential) {
-            Log.i("Noemi", "Phone Verification Completed");
-            signInWithPhoneAuthCredential(phoneAuthCredential);
-        }
-
-        @Override
-        public void onVerificationFailed(@NonNull FirebaseException e) {
-            Log.i("Noemi", e.toString());
-        }
-
-        @Override
-        public void onCodeSent(@NonNull String s, @NonNull PhoneAuthProvider.ForceResendingToken forceResendingToken) {
-            super.onCodeSent(s, forceResendingToken);
-            Log.i("Noemi", "Phone Verification Completed. Code is sent...");
-            FragmentManager dialogFragment = getSupportFragmentManager();
-            EnterCodeDialog enterCodeDialog = new EnterCodeDialog();
-            enterCodeDialog.show(dialogFragment, "dialog");
-            codeSent = s;
-        }
-    };
+    @Override
+    protected void initActivityImpl() {
+        tabLayout = binding.tabLayoutRegister;
+        viewPager = binding.viewPagerRegistration;
 
-    private void verifySignInCode(String verificationCode) {
-        PhoneAuthCredential credential = PhoneAuthProvider.getCredential(codeSent, verificationCode);
-        signInWithPhoneAuthCredential(credential);
-    }
+        fragmentList = new ArrayList<Fragment>(Arrays.asList(new FragmentRegisterPhoneNumber(),new FragmentRegisterName(),new FragmentRegisterBirthday(),new FragmentRegisterGender(),new FragmentRegisterUserType()));
```

Comment:
Good practice :  in loc de new FragmentRegisterPhoneNumber() (si la fel la fiecare fragment), se poate implementa o metoda statica in clasa fragmentului, ceva de genul: ```
public static FragmentRegisterPhoneNumber getInstance() {
return new FragmentRegisterPhoneNumber()
}
``` . Si apoi aici in RegistrationActivity sa faci `FragmentRegisterPhoneNumber.getInstance()`
Asta doar ca sa tinem separat totul legat de fragmente.